### PR TITLE
Microverse Project - Using Bootstrap To Build a Clone of Newsweek.com Website

### PR DESCRIPTION
**Pull request by SunnySparks**
In this project, we designed a clone of the www.newsweek.com home page.

We Implement responsiveness for the screen sizes (Mobile, Tablet, and Desktop) using Bootstrap CSS Framework:

Small screens (lower than 768px)
Medium size screens (Between 768px until 920px) and
Larger screen sizes (greater than 920px)
We have also implemented major CSS styling using bootstrap classes.

### CODE REVIEW

File: css/style.css [hunk 1]
```diff
@@ -0,0 +1,589 @@
...
+}
+
+ul,
+ul li {
+  list-style-type: none;
+}
+
+.navbar-toggler-icon {
+  display: none;
+}
```
Comment (kitpao) [inline css/style.css:32]: - [ ] You have this class but you don't use it in your HTML. Please delete it. Also, you can do the same behavior of display: none; with bootstrap classes (d-none) if you need it

File: css/style.css [hunk 2]
```diff
@@ -0,0 +1,589 @@
...
+
+div.smalldiv {
+  border: none !important;
+}
+
+@media screen and (max-width: 768px) and (min-width: 426px) {
+  .navbar-toggler-icon {
+    display: inline;
+  }
```
Comment (kitpao) [inline css/style.css:248]: - [ ] You have this class but you don't use it in your HTML. Please delete it. Also, you can do the same behavior of display: inline when small screen; with bootstrap classes (d-sm-inline)

File: css/style.css [hunk 3]
```diff
@@ -0,0 +1,589 @@
...
+  .feature-section,
+  .more-stories1 {
+    position: -webkit-sticky;
+    position: sticky;
+    top: 4rem;
+  }
+
+  .feature-cont {
+    align-items: flex-start;
+  }
```
Comment (kitpao) [inline css/style.css:265]: - [ ] You can use bootstrap class for this behavior (d-flex align-items-start). Please use bootstrap classes whenever it is possible

File: css/style.css [hunk 4]
```diff
@@ -0,0 +1,589 @@
...
+
+  .image-thumbnail1 {
+    margin-left: auto;
+  }
+
+  .last-section {
+    flex-direction: row;
+  }
```
Comment (kitpao) [inline css/style.css:273]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: css/style.css [hunk 5]
```diff
@@ -0,0 +1,589 @@
...
+
+  .email-subscription {
+    width: 100%;
+    height: auto;
+  }
+
+  .right-news-cont {
+    display: none;
+  }
```
Comment (kitpao) [inline css/style.css:288]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: css/style.css [hunk 6]
```diff
@@ -0,0 +1,589 @@
...
+
+  .more-stories1 {
+    position: -webkit-sticky;
+    position: sticky;
+    bottom: 1rem;
+    align-self: flex-end;
```
Comment (kitpao) [inline css/style.css:294]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: css/style.css [hunk 7]
```diff
@@ -0,0 +1,589 @@
...
+  }
+
+  #newsletter-large-pic {
+    display: inline;
```
Comment (kitpao) [inline css/style.css:298]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: css/style.css [hunk 8]
```diff
@@ -0,0 +1,589 @@
...
+    background-repeat: no-repeat;
+    background-position: center;
+    background-size: cover;
+    height: 254px;
+  }
+
+  #sticky-footer {
+    margin: 0;
+    padding: 0;
+  }
```
Comment (kitpao) [inline css/style.css:309]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: css/style.css [hunk 9]
```diff
@@ -0,0 +1,589 @@
...
+
+  .footer-copyright,
+  .social {
+    align-self: flex-end;
```
Comment (kitpao) [inline css/style.css:313]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: css/style.css [hunk 10]
```diff
@@ -0,0 +1,589 @@
...
+    margin-right: auto;
+  }
+
+  #footer-content {
+    flex-direction: row;
+  }
```
Comment (kitpao) [inline css/style.css:319]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: css/style.css [hunk 11]
```diff
@@ -0,0 +1,589 @@
...
+
+  .footer-copyright {
+    align-self: flex-end;
+  }
```
Comment (kitpao) [inline css/style.css:323]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: css/style.css [hunk 12]
```diff
@@ -0,0 +1,589 @@
...
+}
+
+@media screen and (max-width: 991px) {
+  #middle-navbar,
+  nav.nav-navigation,
+  nav.navia {
+    display: none;
+  }
+}
```
Comment (kitpao) [inline css/style.css:332]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: css/style.css [hunk 13]
```diff
@@ -0,0 +1,589 @@
...
+
+  header.header-container {
+    position: -webkit-sticky; /* Safari */
+    position: sticky;
+    top: 0;
+  }
+
+  .top-header {
+    padding: 0;
+  }
```
Comment (kitpao) [inline css/style.css:352]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: css/style.css [hunk 14]
```diff
@@ -0,0 +1,589 @@
...
+
+  #middle-navbar {
+    display: block;
+  }
```
Comment (kitpao) [inline css/style.css:356]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: css/style.css [hunk 15]
```diff
@@ -0,0 +1,589 @@
...
+
+  picture {
+    display: inherit;
+  }
+
+  /* This is for the middle navbar */
+
+  .nav-navigation {
+    display: flex;
+    justify-content: space-between;
```
Comment (kitpao) [inline css/style.css:370]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: css/style.css [hunk 16]
```diff
@@ -0,0 +1,589 @@
...
+    height: 55px;
+  }
+
+  .navbar-nav1 {
+    display: flex;
+    flex-wrap: nowrap;
+    align-content: center;
+    align-items: stretch;
```
Comment (kitpao) [inline css/style.css:378]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: css/style.css [hunk 17]
```diff
@@ -0,0 +1,589 @@
...
+    color: black;
+    display: block;
+    padding: 15px 5px;
+    flex-grow: 1;
+  }
+
+  #nav-link-long {
+    padding-left: 0;
+    padding-right: 0;
+  }
```
Comment (kitpao) [inline css/style.css:407]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: css/style.css [hunk 18]
```diff
@@ -0,0 +1,589 @@
...
+    top: -90px;
+  }
+
+  .api {
+    z-index: 100;
+  }
+
+  nav.nav-navigation {
+    display: flex;
+  }
```
Comment (kitpao) [inline css/style.css:433]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: css/style.css [hunk 19]
```diff
@@ -0,0 +1,589 @@
...
+
+  .email-subscription-top {
+    width: 80%;
+    height: auto;
+    display: inline;
+  }
+
+  .form-inline {
+    width: 14%;
+    justify-content: flex-start;
```
Comment (kitpao) [inline css/style.css:479]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: css/style.css [hunk 20]
```diff
@@ -0,0 +1,589 @@
...
+  }
+
+  .form-control {
+    padding: 0;
+  }
```
Comment (kitpao) [inline css/style.css:484]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: css/style.css [hunk 21]
```diff
@@ -0,0 +1,589 @@
...
+
+  .navbar-nav {
+    width: 86%;
+  }
+
+  #coll {
+    margin: 0;
+    display: flex;
+  }
```
Comment (kitpao) [inline css/style.css:493]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: css/style.css [hunk 22]
```diff
@@ -0,0 +1,589 @@
...
+
+  .navbar-expand .navbar-collapse {
+    display: flex !important;
+  }
```
Comment (kitpao) [inline css/style.css:497]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: css/style.css [hunk 23]
```diff
@@ -0,0 +1,589 @@
...
+
+  div.sub-left {
+    background-color: #f4f5f6;
+    height: 350px;
+  }
+
+  div.fn {
+    float: left;
+  }
```
Comment (kitpao) [inline css/style.css:511]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: css/style.css [hunk 24]
```diff
@@ -0,0 +1,589 @@
...
+
+  div.f1 {
+    float: right;
+  }
```
Comment (kitpao) [inline css/style.css:515]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: css/style.css [hunk 25]
```diff
@@ -0,0 +1,589 @@
...
+
+  div.mag-thumb.current {
+    z-index: 6;
+    top: -230px;
+    left: 100px;
+  }
+
+  div.last-section {
+    height: 200px;
+    margin: 0;
```
Comment (kitpao) [inline css/style.css:537]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: css/style.css [hunk 26]
```diff
@@ -0,0 +1,589 @@
...
+  }
+
+  .feature-cont {
+    align-items: flex-start;
+  }
```
Comment (kitpao) [inline css/style.css:542]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: css/style.css [hunk 27]
```diff
@@ -0,0 +1,589 @@
...
+
+  .story-section {
+    position: -webkit-sticky;
+    position: sticky;
+    top: 4rem;
+    align-self: flex-start;
```
Comment (kitpao) [inline css/style.css:548]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: css/style.css [hunk 28]
```diff
@@ -0,0 +1,589 @@
...
+  }
+
+  .feature-section {
+    position: -webkit-sticky;
+    position: sticky;
+    bottom: 1rem;
+    align-self: flex-end;
```
Comment (kitpao) [inline css/style.css:555]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: css/style.css [hunk 29]
```diff
@@ -0,0 +1,589 @@
...
+    margin-left: auto;
+  }
+
+  #news-border {
+    border: none !important;
+  }
+
+  #footer-content {
+    flex-direction: row;
+  }
```
Comment (kitpao) [inline css/style.css:584]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: css/style.css [hunk 30]
```diff
@@ -0,0 +1,589 @@
...
+
+  .footer-copyright {
+    align-self: flex-end;
+  }
```
Comment (kitpao) [inline css/style.css:588]: - [ ] There is a bootstrap class for this behavior, please use bootstrap whenever it is possible.

File: index.html [hunk 31]
```diff
@@ -0,0 +1,976 @@
...
+                        <div class="col-sm-6 mb-2 mt-5 p-1 d-flex justify-content-end">
+                            <button class="btn text-white">Login</button>
+                            <button class="btn bg-warning rounded-pill">Subscribe</button>
+                        </div>
+                    </div>
+                </div>
+                <div class="container-fluid weather">
+                    <div class="api">
+                        <a href="/weather" title="Weather" class="weather-cwidget ml-n3 text-white">
+                            <picture style="display: inherit;">
```
Comment (kitpao) [inline index.html:45]: - [ ] For good coding practices, don't use inline styling, put your styles in your CSS

File: index.html [hunk 32]
```diff
@@ -0,0 +1,976 @@
...
+                                <source srcset="https://g.newsweek.com/img/weather/34.webp" type="image/webp" />
+                                <img src="https://g.newsweek.com/img/weather/34.png" alt="Fair" class="weather-icon" />
+                            </picture>
+                            &nbsp;&nbsp;37°
+                            <span class="weather-country">
+                                Sharjah, AE&nbsp;&nbsp;
+                                <picture style="display: inherit;">
```
Comment (kitpao) [inline index.html:52]: - [ ] For good coding practices, don't use inline styling, put your styles in your CSS

File: css/style.css [hunk 33]
```diff
@@ -0,0 +1,532 @@
...
+}
+
+.fa {
+  color: #656665;
+}
+
+#nl-img-cont {
+  height: 180px;
+  padding-left: 10%;
+}
```
Comment (kitpao) [inline css/style.css:248]: - [ ] A padding left of 25% and a padding top, would be enough for centering this image. Please add it
image

File: index.html [hunk 34]
```diff
@@ -0,0 +1,1383 @@
...
+                <span class="footer-copyright small ml-2 mr-auto mt-4 pr-md-2">© 2020 NEWSWEEK</span>
+                <nav class="bd-highlight d-flex justify-content-end">
+                    <i class="fa fa-facebook-square fa-2x pr-1" aria-hidden="true"></i>
+                    <i class="fa fa-twitter fa-2x pr-1" aria-hidden="true"></i>
+                    <i class="fa fa-instagram fa-2x pr-1" aria-hidden="true"></i>
+                    <i class="fa fa-tumblr-square fa-2x pr-1" aria-hidden="true"></i>
+                    <i class="fa fa-linkedin-square fa-2x pr-1" aria-hidden="true"></i>
+                </nav>
+            </div>
+            <div class="mx-3">
```
Comment (kitpao) [inline index.html:1344]: - [ ] You are still missing this corrections

image

File: index.html [hunk 35]
```diff
@@ -0,0 +1,1383 @@
...
+                                <div class="mt-2"><img src="https://g.newsweek.com/themes/newsweek/favicons/favicon-32x32.png" alt="hello" /></div>
+                                <span class="ml-1"><strong>START YOUR DAY WITH OUR TOP 5 ARTICLES</strong></span>
+                            </div>
+                            <div class="ml-2 mr-2 mt-3">
+                                <input type="text" class="form-text form-control" placeholder="Email address" />
+                                <input type="submit" class="form-submit btn btn-primary btn-block btn-danger mt-3" value="FREE SIGN UP" />
+                            </div>
+                        </div>
+                    </div>
+                    <div class="row mt-3 right-news-cont d-none d-lg-block">
```
Comment (kitpao) [inline index.html:473]: - [ ] You are not using right-news-cont in your css anymore, please take it out

File: index.html [hunk 36]
```diff
@@ -0,0 +1,1383 @@
...
+                                    <input type="text" class="form-text form-control" placeholder="Email address" />
+                                    <input type="submit" class="form-submit btn btn-primary btn-block btn-danger mt-2" value="FREE SIGN UP" />
+                                </div>
+                            </div>
+                        </div>
+                    </div>
+                </article>
+            </div>
+        </div>
+        <footer class="container bg-white pt-5 pt-lg-0" id="sticky-footer">
```
Comment (kitpao) [inline index.html:1332]: - [ ] You are not using id sticky-footer anymore, please delete it

File: index.html [hunk 37]
```diff
@@ -0,0 +1,1383 @@
...
+            <div class="d-flex flex-row pl-3">
+                <div><img src="images/newsweeklogo.png" id="footer-logo" alt="hello"/></div>
+                <span class="footer-copyright small ml-2 mr-auto mt-4 pr-md-2">© 2020 NEWSWEEK</span>
```
Comment (kitpao) [inline index.html:1335]: - [ ] You are not using footer-copyright anymore in your CSS, please take it out

File: index.html [hunk 38]
```diff
@@ -0,0 +1,1383 @@
...
+                                <picture>
+                                    <source srcset="https://g.newsweek.com/img/weather/ic-white-arrow.webp" type="image/webp" />
+                                    <img src="https://g.newsweek.com/img/weather/ic-white-arrow.png" alt="hello" class="weather-arrow" />
+                                </picture>
+                            </span>
+                        </a>
+                    </div>
+                </div>
+            </header>
+            <nav class="nav-navigation d-none d-lg-flex justify-content-lg-between bg-white border border-top-0 border-left-0 border-right-0" id="middle-navbars">
```
Comment (kitpao) [inline index.html:63]: - [ ] You are not using middle-navbars id anymore, please delete it

File: index.html [hunk 39]
```diff
@@ -0,0 +1,1383 @@
...
+                    <li class="nav-item"><a class="nav-link" href="https://www.newsweek.com/topic/coronavirus">Coronavirus</a></li>
+                    <li class="nav-item"><a class="nav-link" href="https://www.newsweek.com/topic/2020-election">2020 Election</a></li>
+                </ul>
+            </nav>
+        </div>
+
+        <main class="container bg-white position-relative pb-2git ">
+            <section class="row">
+                <div class="col-xs-12 col-md-12 col-lg-8 col-xl-9">
+                    <div class="row d-md-flex feature-cont align-items-md-start">
```
Comment (kitpao) [inline index.html:99]: - [ ] You are not using feature-cont class, please delete it

File: index.html [hunk 40]
```diff
@@ -0,0 +1,1383 @@
...
+                <div class="w-30 p-0 h-auto float-left">
+                    <img src="images/Newsweek_Logo.svg" alt="" class="header-logo">
+                </div>
+                <div class="w-70 float-right">
+                    <div class="float-right mr-5 mt-1 align-self-end d-sm-inline d-md-inline d-lg-none navbar-toggler text-white">
+                        <div class="navbar-toggler-icon text-white"></div>
+                        <div class="navbar-collapse"></div>
+                    </div>
+                    <ul class="top-navbar mt-2 float-right pr-2">
+                        <li class="btn btn-sm text-white d-none d-sm-inline d-md-inline"><a href="#" class="text-white">Login</a></li>
```
[the remark this answers is not in the document]
Comment (happiguru) [reply index.html:25]: Remove the conflicting class. Resolved

File: index.html [hunk 41]
```diff
@@ -0,0 +1,1383 @@
...
+                                    <img class="w-100 img-fluid" src="https://d.newsweek.com/en/full/1610532/house-minority-leader-kevin-mccarthy.jpg?w=397&h=265&q=90&f=a2d13c7372fd3d222866ae61de2bb167" alt="t" />
+                                </div>
+                                <div class="caption pt-1"><a href="https://www.newsweek.com/us">U.S.</a></div>
+                                <header class="w-100 mt-n4">
+                                    <h3 class="ml-0">
+                                        <a href="https://www.newsweek.com/mccarthy-gop-election-2020-1518540" title="" class="text-dark"> <strong>House GOP Leader Says Party May Never Win Election Again if Dems Sweep 2020</strong></a>
+                                    </h3>
+                                </header>
+                                <p class="ml-0">The House minority leader said Democrats would "change the rules of the game" were they to win November's election.</p>
+                            </article>
```
[the remark this answers is not in the document]
Comment (happiguru) [reply index.html:169]: Resolved this issue

File: css/style.css (hunk 33 above)
Comment (happiguru) [reply css/style.css:248]: Issue fixed

File: index.html (hunk 35 above)
Comment (happiguru) [reply index.html:473]: Removed unused class

File: index.html (hunk 37 above)
Comment (happiguru) [reply index.html:1335]: Removed unused class

File: index.html (hunk 39 above)
Comment (happiguru) [reply index.html:99]: Removed the unused class

File: index.html (hunk 34 above)
Comment (happiguru) [reply index.html]: Issue fixed

File: index.html (hunk 36 above)
Comment (happiguru) [reply index.html:1332]: Removed unused id

**Pull request merged**